src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -649,6 +649,23 @@ public function dumpFile($filename, $content)
     }
 
     /**
+     * Atomically appends content to an existing file.
+     *
+     * @param string $filename The file for which to append content
+     * @param string $content  The content to append
+     *
+     * @throws IOException If the file is not writable
+     */
+    public function appendToFile($filename, $content)
+    {
+        if (false !== $originalContent = @file_get_contents($filename)) {
```

Comment:
So what about a public `writeFile($filename, $content[, $append = false])` then? Using `LOCK_EX` by design and let `dumpFile` forward? That to me would be the simplest implementation.

But a private function in the middle allowing this API (`appendToFile` and `dumpFile`) is also fine :+1:  i guess.